src/main/java/org/apache/brooklyn/cloudfoundry/utils/LocalResourcesDownloader.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.cloudfoundry.utils;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.brooklyn.util.core.ResourceUtils;
+import org.apache.brooklyn.util.os.Os;
+import org.apache.brooklyn.util.text.Strings;
+import org.apache.commons.io.FileUtils;
+
+import com.google.common.base.Throwables;
+
+public class LocalResourcesDownloader {
```

Comment:
Actually, my experience says that the driver will no add a lot of complexity and it will useful to allow different locations to be supported by an entity. In any case, this PR does not modified the `VanillaCloudFoundryApplication` entity. 

But in any case, I infer that `CloudFoundryPaasLocation` (or `CloudFoundryPaasClient`) should receive an artifact uri which will be ready to be pushed to Cloud Foundry and the entity (or the driver) has to be in charge of provides a valid uri. 

So, my proposal. `deploy` method will receive an local artifact uri ready to be push to the platform (instead of worrying about downloading the artifacts and save it in the temp folder). Then, I will provides a `Driver`driven approach in the next PR where I will integrate the `CloudFoundryLocation`and the `VanillaCloudFoundryApplication`.
Then, this PR will allow to decide about `CloudFoundryPaasLocation`interface.

WDYT?